data/src/test/java/org/apache/iceberg/data/orc/TestGenericData.java

```diff
@@ -54,13 +54,23 @@
 
 public class TestGenericData extends DataTest {
 
+  @Override
+  protected boolean supportsRequiredTypes() {
```

Comment:
if possible, maybe add some Javadoc to explain why the behavior is different for ORC and hence have to disable the `testWriteNullValueForRequiredType` from the base class

Reply:
Added. The Orc writers also run this test now, as it is expected for them to support writing null values for required fields.